lib/colorls/core.rb

```diff
@@ -164,20 +164,29 @@ def display_report
         .colorize(@colors[:report])
     end
 
+    def format_mode(rwx, special, char)
+      r = 'r'.colorize(@colors[:read]).freeze
+      w = 'w'.colorize(@colors[:write]).freeze
+      x = 'x'.colorize(@colors[:exec]).freeze
+      na = '-'.colorize(@colors[:no_access]).freeze
```

Comment:
Well, I think the `freeze` calls really have no effect here. I wanted to reduce string allocations, but this only works for literals.

How about making these constants attributes of the class? That way they would be only initialized once, not for every single file. What do you think, @athityakumar?